### PR TITLE
Refactor: Strait jacket to extended item

### PR DESCRIPTION
**Pull request by SandrineBC**
# Summary
Refactored the strait jacket to use the extended item script.

### CODE REVIEW

File: BondageClub/Scripts/Asset.js [hunk 1]
```diff
@@ -150,6 +150,7 @@ function AssetAdd(NewAsset) {
 		DynamicBeforeDraw: (typeof NewAsset.DynamicBeforeDraw === 'boolean') ? NewAsset.DynamicBeforeDraw : false,
 		DynamicAfterDraw: (typeof NewAsset.DynamicAfterDraw === 'boolean') ? NewAsset.DynamicAfterDraw : false,
 		DynamicScriptDraw: (typeof NewAsset.DynamicScriptDraw === 'boolean') ? NewAsset.DynamicScriptDraw : false,
+		InheritColor: NewAsset.InheritColor,
```
Comment (ace-1331) [inline BondageClub/Scripts/Asset.js:153]: You're re-adding this line, probably due to a merge conflict

**Review by ace-1331 — CHANGES_REQUESTED** [review]

File: BondageClub/Screens/Inventory/ItemArms/StraitJacket/StraitJacket.js [hunk 2]
```diff
@@ -1,75 +1,85 @@
...
+ * The NPC dialog is for what the NPC says to you when you make a change to their restraints - the dialog lookup is on a 
+ * per-NPC basis. You basically put the "AssetName" + OptionName in there to allow individual NPCs to override their default 
+ * "GroupName" dialog if for example we ever wanted an NPC to react specifically to having the restraint put on them. 
+ * That could be done by adding an "AssetName" entry (or entries) to that NPC's dialog CSV
+ * @param {Character} C - The NPC to whom the restraint is applied
+ * @param {Option} Option - The chosen option for this extended item
+ * @returns {void} - Nothing
+ */
+function InventoryItemArmsStraitJacketNpcDialog(C, Option) {
+	C.CurrentDialog = DialogFind(C, "RoundPiercingNPCReaction" + Option.Name, "ItemNipplesPiercings");
```
Comment (ace-1331) [inline BondageClub/Screens/Inventory/ItemArms/StraitJacket/StraitJacket.js:84]: Still using the piercing texts here

File: BondageClub/Assets/Female3DCG/Female3DCG.js [hunk 3]
```diff
@@ -1464,19 +1464,19 @@ var AssetFemale3DCG = [
...
+			{ Name: "StraitJacket", Fetish: ["Leather"], Value: 150, Difficulty: 6, SelfBondage: 8, Time: 35, RemoveTime: 20, AllowLock: true, DefaultColor: "#A0A0A0", Hide: ["Cloth", "ItemNipplesPiercings"], HideItem: ["ClothLowerSkirt1", "ClothLowerSkirt2", "ClothLowerSkirt3", "ClothLowerTennisSkirt1", "ClothLowerGown2Skirt", "ClothLowerWaspie1", "ClothLowerWaspie2", "ClothLowerWaspie3", "ClothLowerLatexSkirt1", "ClothLowerLatexSkirt2", "ClothLowerClothSkirt1", "ClothLowerChineseSkirt1", "ClothLowerAdmiralSkirt", "ClothLowerJeanSkirt", "ClothLowerPencilSkirt"], SetPose: ["BackElbowTouch"], Effect: ["Block", "Prone"], Block: ["ItemNipples", "ItemNipplesPiercings", "ItemTorso", "ItemBreast", "ItemHands"], Extended: true, SelfUnlock: false, HasType: false },
```
Comment (ace-1331) [inline BondageClub/Assets/Female3DCG/Female3DCG.js:1479]: Missing allowtypes, so it's not saving properly

**Review by ace-1331 — CHANGES_REQUESTED** [review]

File: BondageClub/Assets/Female3DCG/Female3DCG.js (hunk 3 above)
Comment (SandrineBC) [reply BondageClub/Assets/Female3DCG/Female3DCG.js:1479]: That AllowTypes issue must have been there for ages...

**Review by ace-1331 — APPROVED** [review]

**Comment by ace-1331** [comment]
After merging #1336, I'm going through PRs to check what might conflict/be missing.

When you resolve the conflicts, please make sure the items retain the proper SetPose/AllowPose/AllowActivePose, etc. set in 1336 as it appears this PR might change some of the same lines.

**Comment by SandrineBC** [comment]
Looks good after the merge

File: BondageClub/Assets/Female3DCG/Female3DCG.js [hunk 4]
```diff
@@ -1478,7 +1478,7 @@ var AssetFemale3DCG = [
...
+			{ Name: "StraitJacket", Value: 150, Difficulty: 6, SelfBondage: 8, Time: 35, RemoveTime: 20, AllowLock: true, DefaultColor: "#A0A0A0", Hide: ["Cloth", "ItemNipplesPiercings"], HideItem: ["ClothLowerSkirt1", "ClothLowerSkirt2", "ClothLowerSkirt3", "ClothLowerTennisSkirt1", "ClothLowerGown2Skirt", "ClothLowerWaspie1", "ClothLowerWaspie2", "ClothLowerWaspie3", "ClothLowerLatexSkirt1", "ClothLowerLatexSkirt2", "ClothLowerClothSkirt1", "ClothLowerChineseSkirt1", "ClothLowerAdmiralSkirt", "ClothLowerJeanSkirt", "ClothLowerPencilSkirt"], SetPose: ["BackElbowTouch"], Effect: ["Block", "Prone"], AllowType: ["Normal", "Snug", "Tight"], Block: ["ItemNipples", "ItemNipplesPiercings", "ItemTorso", "ItemBreast", "ItemHands"], Extended: true, SelfUnlock: false, HasType: false },
```
Comment (ace-1331) [inline BondageClub/Assets/Female3DCG/Female3DCG.js:1481]: I assume removing the fetish was not intended here?

**Review by ace-1331 — CHANGES_REQUESTED** [review]

File: BondageClub/Assets/Female3DCG/Female3DCG.js (hunk 4 above)
Comment (SandrineBC) [reply BondageClub/Assets/Female3DCG/Female3DCG.js:1481]: In fact it was, as the leather fetish made no sense. We have the leather strait jacket for this fetish, so I removed it on the canvas version. Perhaps we could introduce a 'medical' fetish at one point?

**Review by ace-1331 — APPROVED** [review]

**Review by elliesec — APPROVED** [review]

**Pull request merged**